src/server/object_services/object_server.js

```diff
@@ -1950,6 +1992,7 @@ async function _complete_object_multiparts(obj, multipart_req) {
     await Promise.all([
         context.parts_updates.length && MDStore.instance().update_parts_in_bulk(context.parts_updates),
         used_multiparts_ids.length && MDStore.instance().update_multiparts_by_ids(used_multiparts_ids, undefined, { uncommitted: 1 }),
+        used_multiparts_ids.length && MDStore.instance().update_object_by_id(obj._id, { num_multiparts: used_multiparts_ids.length }),
```

Comment:
When used_multiparts_ids.length is 0, should we really leave this field unset?

Reply:
@guymguym, I see no reason why.
Am I missing something?

Reply:
because it will cause us to recalculate it every time

Reply:
isn't that what you do when it's missing?

Reply:
@guymguym I guess if used_multiparts_ids.length is 0, then there are no "multiparts". Could you please describe a scenario when used_multiparts_ids.length is 0 and it causes us to recalculate it every time?